staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -128,8 +129,9 @@ func newApfServerWithSingleRequest(t *testing.T, decision mockDecision) *httptes
 			t.Errorf("execute should not be invoked")
 		}
 		// atomicReadOnlyExecuting can be either 0 or 1 as we test one request at a time.
-		if decision != decisionSkipFilter && atomicReadOnlyExecuting != 1 {
-			t.Errorf("Wanted %d requests executing, got %d", 1, atomicReadOnlyExecuting)
+		executing := atomic.LoadInt32(&atomicReadOnlyExecuting)
```

Comment:
This would fix the panic itself.

But given there is a race, with this change the test might still fail (instead of panicing it may now be just failing).

Reply:
BTW, the race detected involved old line 179, not old line 131.

Reply:
True - but the mechanism is exactly the same in all of those.